### ISSUE TITLE
Fix "CItemAirBox" flight to the moon!

### ISSUE THREAD

**Issue by StevenKal** [s1lentq/ReGameDLL_CS#681]
The custom & new "CItemAirBox" entity is cool, but I have spotted an issue with it when the fly is active ("m_flyup" & "m_delay" > 0).
As you can [here](https://www.amxmod.net/images/misc/CItemAirBox_issue_-_01.jpg), the items keep flying in the air without stopping, while the item is supposed to keep doing the "yoyo" (up & down).
This happens when they are "Restarted" and only if we picked them up before.
This is due to the fact, once picked up by a client, the think function is set to "NULL" (in the "CItemAirBox::Touch") and the "m_flyup" member is, when the fly is active, constantly altered to be positive or negative (in the "CItemAirBox::MoveUp").
So as result, when it is restarted, the "m_flyup" can be negative and the "CItemAirBox::MoveUp" function is not put back, so the item keeps moving with its previous velocity.

As a fix, I recommend to reset the velocity to 0 (z only, or, why not all axes) in the "CItemAirBox::Restart" (right below "UTIL_SetOrigin"), then also in "CItemAirBox::Touch", right below the "SetThink", to add this:
if(m_flyup < 0) {
  m_flyup = -m_flyup;
}

PS: Sorry I can not submit PR this is bugged under XP!

**Comment by RauliTop** [comment]
It's like Bitcoin. Do you know

**Comment by StevenKal** [comment]
Yeah! I do not use yet such cryptomoney, but that I can say is "I was pretty surprised when I saw the recent value of 1 bitcoin not so long ago". Diamond money!
